tests/conftest.py

```diff
@@ -46,37 +47,163 @@ def matlab_engine(refactored = False):
+                    contents = file.readlines()  # Read the file as a list of lines
+                return contents
+        
+        # If no .txt file is found, raise an error
+        raise FileNotFoundError(f"No .txt file found in folder: {folder}")
+
+    return _find_text_file_in_folder
+
+@pytest.fixture
+def extract_section_between_keywords():
+    """
+    Fixture to extract a section from a results txt file between two keywords.
+
+    Args:
+        data (list): The contents of the file as a list of lines.
+        start_keyword (str): The keyword to start extracting from.
+        end_keyword (str, optional): The keyword to stop extracting before. 
+                                     Defaults to None, which means take all text onward from the start keyword.
+
+    Returns:
+        function: A function that takes data (list of lines), start_keyword (str), 
+                  and end_keyword (str, optional), and returns the lines between the 
+                  start keyword and the end keyword, exclusive of the end keyword.
+    """
+    def _extract(data, start_keyword, end_keyword=None):
+        # Find the index of the first occurrence of the start keyword
+        start_idx = -1
+        for i, line in enumerate(data):
+            if line.strip().startswith(start_keyword):
+                start_idx = i
+                break  # Stop after finding the first occurrence
+
+        if start_idx == -1:
+            raise ValueError(f"No section starting with '{start_keyword}' found in the file")
+
+        # If end_keyword is provided, find its index
+        end_idx = len(data)  # Default to end of file if end_keyword is None
+        if end_keyword:
+            for i, line in enumerate(data[start_idx + 1:], start=start_idx + 1):
+                if line.strip().startswith(end_keyword):
+                    end_idx = i
+                    break  # Stop after finding the first occurrence
+
+        # Extract the section between the start and end keywords
+        section = data[start_idx + 1:end_idx]  # Exclusive of both start and end keywords
+
+        return section
+
+    return _extract
+
+def process_std_out(std_out):
+    """
+    Process standard output from a StringIO object into a list of lines.
+
+    Args:
+        std_out (StringIO): The StringIO object containing standard output.
+
+    Returns:
+        list: A list of lines from the standard output.
+    """
+    output = std_out.getvalue()
+    output_lines = output.splitlines()
+    return output_lines
+
+def compare_text_blocks(text1, text2):
+    """
+    Compare two blocks of text after stripping whitespace.
+
+    Args:
+        text1 (str): The first block of text.
+        text2 (str): The second block of text.
 
-    return str(input_folder), str(output_folder)
+    Returns:
+        bool: True if the stripped text blocks are identical, False otherwise.
+    """
+    return text1.strip() == text2.strip()
```

Comment:
May want to handle the case for removing new lines if comparing longer output: e.g. `.replace('\n', '')`